vms/platformvm/warp/signature.go

```diff
@@ -23,6 +24,17 @@ var (
 	ErrParseSignature     = errors.New("failed to parse signature")
 )
 
+// VerifiableValidatorState allows the lookup of validator sets on specified subnets at the
+// requested P-chain height as well as the retrieval of subnet ids.
+type VerifiableValidatorState interface {
```

Comment:
We don't need to disambiguate across `validators.State` because this is its own package - can we just call this `ValidatorState` or `Validators` or something like that?

Reply:
I was thinking we should add the necessary functions to the existing `ValidatorState` interface here: https://github.com/ava-labs/avalanchego/blob/ab4619873028fd975dfa6e89e53e555dd53e0cdb/vms/platformvm/warp/validator.go#L32

The downside is ofc that the functions we're adding to it are not strictly needed.

Perhaps a better alternative is to modify the function signature of `*BitSetSIgnature.Verify(...)` to take in an already calculated canonical validator set and provide a helper function that converts from `validators.State` to the canonical validator set input that's required.